fiona/collection.py

```diff
@@ -284,12 +284,12 @@ def set_tags(self, tags, ns=None):
         """
         if _GDAL_VERSION_TUPLE.major < 2:
             raise GDALVersionError(
-                "set_tags requires GDAL 2+, fiona was compiled "
+                "update_tags requires GDAL 2+, fiona was compiled "
                 "against: {}".format(_GDAL_RELEASE_NAME)
             )
         if not isinstance(self.session, WritingSession):
-            raise DataIOError("Unable to set tags as not in writing mode.")
-        return self.session.set_tags(tags, ns=ns)
+            raise DataIOError("Unable to update tags as not in writing mode.")
```

Comment:
@snowman2 I feel like I'm getting dangerously close to the "while you're at it..." review anti-practice, forgive me, but can you create a new fiona.errors.UnsupportedOperation exception to use here (like rasterio's https://github.com/mapbox/rasterio/blob/master/rasterio/errors.py#L103 and like the one in Python's io module)?

Reply:
Sounds like a logical addition. I pushed the change.